modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerRecoveryTest.java

```diff
@@ -101,6 +104,44 @@ void testRecoveryCatalogVersionTimestamps() throws Exception {
         assertThat(catalogManager.activeCatalogVersion(time1), equalTo(catalogVersion1));
     }
 
+    @Test
+    void testRecoveryCatalogAfterCompaction() throws Exception {
```

Comment:
Don't know where to put this note - CatalogValidationSchemasSource still contains a TODO with a link to this issue.

Also this test is flaky (I'm sure you know).

Reply:
Thanks. Fixed.